### PR TITLE
Refactor generator to generate time-based traces

### PR DESCRIPTION
**Pull request by happyandslow**
This commit adds 
- Leading timestamp for each concurrent dispatch
- Force include input/output token lengths in the dataset
- Option to store as .jsonl

### CODE REVIEW

File: benchmarks/generator/workload_generator.py [hunk 1]
```diff
@@ -180,64 +180,64 @@ def generate_from_azure_csv(file_path: str,
...
-    parser.add_argument('--num-requests', type=int, default=10000, help='Number of requests in total')
-    parser.add_argument('--group-interval-seconds', type=int, default=1, help='Grouping interval seconds')
-    parser.add_argument('--trace-type', type=str, required=True, default="synthetic", help='Type of trace consumed')
-    parser.add_argument('--trace-file', type=str, required=False, default=None, help='File containing trace CSV')
-    parser.add_argument('--model', type=str, required=False, default="meta-llama/Llama-2-7b-hf", help='Target model tokenizer')
-    parser.add_argument('--output', type=str, required=False, default="output/output", help='Output path')
+    parser.add_argument('--prompt-file', type=str, required=True, help='File containing prompts.')
+    parser.add_argument('--num-prompts', type=int, default=100, help='Number of prompts to sample.')
+    parser.add_argument('--group-interval-seconds', type=int, default=1, help='Grouping interval seconds.')
+    parser.add_argument('--trace-type', type=str, required=True, default="synthetic", help='Type of trace consumed.')
```
Comment (Jeffwan) [inline benchmarks/generator/workload_generator.py]: I suggest to have all `--trace-type` listed here.

File: benchmarks/generator/workload_generator.py [hunk 2]
```diff
@@ -180,64 +180,64 @@ def generate_from_azure_csv(file_path: str,
...
+    parser.add_argument('--trace-file', type=str, required=False, default=None, help='File containing trace CSV.')
```
Comment (Jeffwan) [inline benchmarks/generator/workload_generator.py]: Let's add more details in the `help`.  what kind of trace-type needs such file etc.

File: benchmarks/generator/workload_generator.py [hunk 3]
```diff
@@ -180,64 +180,64 @@ def generate_from_azure_csv(file_path: str,
...
+    parser.add_argument('--model', type=str, required=False, default="meta-llama/Llama-2-7b-hf", help='Target model tokenizer.')
```
Comment (Jeffwan) [inline benchmarks/generator/workload_generator.py]: let's change the default one to something else. like https://huggingface.co/Qwen/Qwen2.5-Coder-7B-Instruct. llama7b can not be publicly accessible

File: benchmarks/generator/workload_generator.py [hunk 4]
```diff
@@ -180,64 +180,64 @@ def generate_from_azure_csv(file_path: str,
...
+    parser.add_argument('--output', type=str, required=False, default="output/output", help='Output path to the workload.')
+    parser.add_argument('--interval-ms', type=int, required=False, default=1000, help='Granularity of request injection interval in milliseconds.')
+    parser.add_argument('--duration-ms', type=int, default=60000, help='Duration of the trace generated.')
```
Comment (Jeffwan) [inline benchmarks/generator/workload_generator.py:213]: suggest to use `second` as the time unit. for duration, normally, we at least generate 120-600 etc. won't be any case less than 1s

File: benchmarks/generator/workload_generator.py [hunk 5]
```diff
@@ -180,64 +180,64 @@ def generate_from_azure_csv(file_path: str,
...
-    parser.add_argument('--trace-file', type=str, required=False, default=None, help='File containing trace CSV')
-    parser.add_argument('--model', type=str, required=False, default="meta-llama/Llama-2-7b-hf", help='Target model tokenizer')
-    parser.add_argument('--output', type=str, required=False, default="output/output", help='Output path')
+    parser.add_argument('--prompt-file', type=str, required=True, help='File containing prompts.')
+    parser.add_argument('--num-prompts', type=int, default=100, help='Number of prompts to sample.')
+    parser.add_argument('--group-interval-seconds', type=int, default=1, help='Grouping interval seconds.')
+    parser.add_argument('--trace-type', type=str, required=True, default="synthetic", help='Type of trace consumed.')
+    parser.add_argument('--trace-file', type=str, required=False, default=None, help='File containing trace CSV.')
+    parser.add_argument('--model', type=str, required=False, default="meta-llama/Llama-2-7b-hf", help='Target model tokenizer.')
+    parser.add_argument('--output', type=str, required=False, default="output/output", help='Output path to the workload.')
```
Comment (Jeffwan) [inline benchmarks/generator/workload_generator.py]: `output/output` -> `output`?

File: benchmarks/generator/workload_generator.py (hunk 1 above)
Comment (happyandslow) [reply benchmarks/generator/workload_generator.py]: Fixed in new commit.

File: benchmarks/generator/workload_generator.py (hunk 2 above)
Comment (happyandslow) [reply benchmarks/generator/workload_generator.py]: Fixed in new commit.

File: benchmarks/generator/workload_generator.py (hunk 3 above)
Comment (happyandslow) [reply benchmarks/generator/workload_generator.py]: Fixed in new commit.

File: benchmarks/generator/workload_generator.py (hunk 5 above)
Comment (happyandslow) [reply benchmarks/generator/workload_generator.py]: Fixed in new commit.

**Pull request merged**